src/xtgeo/xyz/polygons.py

```diff
@@ -164,56 +147,59 @@ def dhname(self):
 
         .. versionadded:: 2.1
         """
-        if isinstance(self._dhname, str):
-            if self._dhname not in self._df.columns:
-                self._dhname = None
-
         return self._dhname
 
     @dhname.setter
-    def dhname(self, mydhname):
-        if isinstance(mydhname, str):
-            self._dhname = mydhname
-        else:
-            raise ValueError("Wrong type of input to dhname; must be string")
+    def dhname(self, value):
+        super()._check_name(value)
+        self._dhname = value
 
     @property
     def tname(self):
         """Returns or set the name of the cumulative total length column if it exists.
 
         .. versionadded:: 2.1
         """
-        if isinstance(self._tname, str):
-            if self._tname not in self._df.columns:
-                self._tname = None
-
         return self._tname
 
     @tname.setter
-    def tname(self, mytname):
-        if isinstance(mytname, str):
-            self._tname = mytname
-        else:
-            raise ValueError("Wrong type of input to tname; must be string")
+    def tname(self, value):
+        super()._check_name(value)
+        self._tname = value
 
     @property
     def dtname(self):
         """Returns or set the name of the delta total length column if it exists.
 
         .. versionadded:: 2.1
         """
-        if isinstance(self._dtname, str):
-            if self._dtname not in self._df.columns:
-                self._dtname = None
-
         return self._dtname
 
     @dtname.setter
-    def dtname(self, mydtname):
-        if isinstance(mydtname, str):
-            self._dtname = mydtname
-        else:
-            raise ValueError("Wrong type of input to dhname; must be string")
+    def dtname(self, value):
+        super()._check_name(value)
+        self._dtname = value
+
+    @property
+    def dataframe(self):
+        """Returns or set the Pandas dataframe object."""
+        return self._df
+
+    @dataframe.setter
+    def dataframe(self, df):
```

Comment:
Currently this setter resides in `Polygons` only, as neither `Points` nor `XYZ` had this setter up front. We could move this as well to `XYZ`, which would add to the interface of `Points`

Reply:
The reason the test failed was because the dataframe setter wasn't set properly. I updated the code now, having to explicitly call the inherited class' attribute instead of using the `super` format. Also necessary to go by the attributes' `fset` function. It looks a bit off maybe, and my pylint is complaining with:
>Function 'dataframe' has no 'fset' memberpylint(no-member)

But the functionality is otherwise sound. I can do some more checking whether using `fset` is considered pythonic enough..